### ISSUE TITLE
Bug: 0.7.0 binary built with go1.14 incorrectly panics on certain Linux kernels

### ISSUE THREAD

**Issue by stealthybox** [bitnami/bitnami-docker-external-dns#4]
### problem

The problem is introduced in the `0.7.0` bitnami image and fixed in `0.7.1`.

The `0.7.0` rolling tag binaries are built with `go1.14` which can cause external-dns to panic with a false positive for the [go/wiki/LinuxKernelSignalVectorBug](https://github.com/golang/go/wiki/LinuxKernelSignalVectorBug) on patched kernels.

The `0.7.1` rolling tag binaries are built with `go1.14.1` which fixes this problem because it includes [golang/go#37807](https://github.com/golang/go#37807).

```shell
docker run --rm --entrypoint sh bitnami/external-dns:0.7.1-debian-10-r41 -c 'cat "$(which external-dns)" | grep -ao "go1\....."' | head -n1
go1.14.1

docker run --rm --entrypoint sh bitnami/external-dns:0.7.0-debian-10-r0 -c 'cat "$(which external-dns)" | grep -ao "go1\....."' | head -n1
go1.14go
```

These go versions do not match some of the upstream builds.

The upstream external-dns container images were intended to be built with `go1.14`, but they actually were incorrectly built with `go1.13.8` and `go1.13.9` so these upstream binaries do not throw false-positives on certain patched kernels and have a hidden memory-corruption bug on affected kernels.

The problem is described in more detail in this upstream patch description: [kubernetes-sigs/external-dns#1554](https://github.com/kubernetes-sigs/external-dns/pull/1554)

### solution

I propose we build new binaries for the rolling-tag revisions for bitnami images `0.7.0` and `0.7.1` with either `go1.14.1` or `go1.14.2`.

I don't know how these binaries are built, but they appear to be magically downloaded and upnpacked by https://github.com/bitnami/bitnami-docker-external-dns/blob/4ac7a1b/0/debian-10/Dockerfile#L6.
( is this documented somewhere? )

__________________
```
Kubernetes 1.16.8
Docker 18.09.7

cat /etc/lsb-release; uname -srvp
DISTRIB_ID=Ubuntu
DISTRIB_RELEASE=19.10
DISTRIB_CODENAME=eoan
DISTRIB_DESCRIPTION="Pop!_OS 19.10"
Linux 5.3.0-7648-generic #41~1586789791~19.10~9593806-Ubuntu SMP Mon Apr 13 17:50:40 UTC  x86_64
```

**Comment by miguelaeh** [comment]
Hi @stealthybox ,
Thank you very much for reporting this issue.
Regarding:
> is this documented somewhere?

We use internally our tool https://github.com/bitnami/blacksmith to build the applications. Nevertheless, the code we use internally to build the applications using blacksmith is not public.

Regarding the issue, you said it is already fixed on the `0.7.1` version, our internal pipeline will not publish the version `0.7.0` anymore as we always publish the latest versions (and stable versions).
So I would like to clarify this:
> I propose we build new binaries for the rolling-tag revisions for bitnami images 0.7.0 and 0.7.1 with either go1.14.1 or go1.14.2.

If the issue is fixed in the version: `0.7.1`, do you think there is something more we can do with this version?

**Comment by stealthybox** [comment]
> If the issue is fixed in the version: 0.7.1, do you think there is something more we can do with this version?

Reading my post again, I don't see what I was implying about `v0.7.1`.
We could only improve `0.7.1` by building with `go1.14.2` or later which is not useful for this issue.
The go version doesn't match upstream, but it would regress on the kernel-related bug to build with go1.13.x.
_____

If the pipeline cannot publish a newer build revision for `v0.7.0`, is there a way to mark it as broken?

Given how sophisticated bitnami's revision system is and that binaries are built from source, it seems like it should be possible to:
1. deprecate a build
2. publish a new revision for an older software release

If neither of those are possible, we could close this bug with `v0.7.0` as wont-fix.

Though it might be important to get the team members working with blacksmith and the related publishing pipelines to consider these two points.

Thanks for the thoughtful response!

**Comment by miguelaeh** [comment]
Hi @stealthybox ,
I am afraid we cannot mark the version as a bug, this is because we provide the latest versions of all the apps, for example, the external-dns major version is the `0`, that means that we will publish new versions without deleting the old ones from `0.0.1` to `0.x.x`. Then, when the major `1` appears, we will start publishing the `1.x.x`, and when the `0` is no more maintained we will stop publishing it.
So basically, we publish always the latest versions for each branch (major version) of the application, and we let the users the task to updates their images.
You can see here https://hub.docker.com/r/bitnami/external-dns/tags all the tags that we have for this image
